packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -2114,11 +2112,8 @@ class JavaGenerator extends Generator {
     if (props.some((prop) => prop.fieldJavaType !== prop.paramJavaType)) {
       this.code.line('@SuppressWarnings("unchecked")');
     }
-    const constructorArgs = props
-      .map((prop) => `final ${prop.paramJavaType} ${prop.fieldName}`)
-      .join(', ');
     this.code.openBlock(
-      `protected ${INTERFACE_PROXY_CLASS_NAME}(${constructorArgs})`,
+      `protected ${INTERFACE_PROXY_CLASS_NAME}(final ${BUILDER_CLASS_NAME} builder)`,
```

Comment:
Certainly a bit non-traditional use of a builder, but the alternative would require the fields to be mutable. Does seem fine given that this is an internal class.